src/password-field.tsx

```diff
@@ -69,23 +74,29 @@ const PasswordField: React.FC<PasswordFieldProps> = ({
     ...rest
 }) => {
     const [isVisible, setIsVisible] = React.useState(false);
+    const caretPositionRef = React.useRef<number>();
```

Comment:
does TS allow this? if it's a ref of type number, shouln't you give it a intial value of type number?

Reply:
TS doesn't complaint because:
```
function useRef<T = undefined>(): MutableRefObject<T | undefined>;
```
I'll initialize with 0 anyway